bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -314,7 +316,7 @@ public int getTimestamp() {
      * @return the Date
      */
     public Date getDate() {
-        return new Date(timestamp * 1000L);
+        return new Date(((long)timestamp & 0xFFFFFFFFL) * 1000L);
```

Comment:
The cast to long is unnecessary, as the bitwise operator will automatically promote timestamp to along in order to mask it with a long.

https://docs.oracle.com/javase/specs/jls/se11/html/jls-15.html#jls-15.22.1
https://docs.oracle.com/javase/specs/jls/se11/html/jls-5.html#jls-5.6.2

Reply:
I have removed the cast.